jlm/llvm/opt/reduction.cpp

```diff
@@ -111,11 +113,39 @@ enable_binary_reductions(rvsdg::Graph & graph)
   nf->set_mutable(true);
   nf->set_reducible(true);
 }
+#endif
+
+static bool
+NormalizeLoadNode(LoadNonVolatileNode & node)
+{
+  static auto LoadNormalizationSequence =
+      rvsdg::NodeNormalizationSequence<LoadNonVolatileOperation>(
+          { LoadMuxNormalization, LoadAllocaNormalization });
```

Comment:
Does this vector live long enough to still be valid when the normalization is actually called?

I think you would have to place this vector as a static on the line above, and pass in a reference?

Reply:
You are right. The vector would need to be placed as state on the line above. Minor issue though.

Reply:
Should not be an issue any longer.